be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -1569,6 +1569,11 @@ Status SegmentIterator::_build_context(ScanContext* ctx) {
         }
     }
 
+    if (read_indexes.size() > output_schema().num_fields()) {
+        return Status::Corruption(
```

Comment:
not sure if there is more info can be logged for debugging, e.g. the segment filepath or the size of the read_index  and the size of fields in schema.

Reply:
This crash may only occur in certain situations in cross-cluster replication.  The problem is clear and does not need to debug. This pr is just a protection for be.

Reply:
In a normal normal filter unused column optimization, the read_index size is likely to be smaller than the output column.